domains/pallets/messenger/src/messages.rs

```diff
@@ -72,6 +72,15 @@ impl<T: Config> Pallet<T> {
                         .latest_response_received_message_nonce,
                 };
                 Outbox::<T>::insert((dst_chain_id, channel_id, next_outbox_nonce), msg);
+                OutboxMessageCount::<T>::try_mutate(
```

Comment:
If this error happens, the message has already been inserted into storage. Does the error cause a rollback of storage?
(I understand the error is unlikely.)

If not, we could increment the message count before we insert the message.

I also have similar questions about the nonce increment below.